src/noit_lmdb_tools.c

```diff
@@ -342,16 +343,16 @@ void noit_lmdb_resize_instance(noit_lmdb_instance_t *instance)
   uint64_t new_mapsize;
 
   /* prevent new transactions on the write side */
+
+  uint64_t initial_generation = noit_lmdb_get_instance_generation(instance);
```

Comment:
It isn't safe to call this here. This `initial_generation` should be an argument to the resize function.  The called should call the `noit_lmdb_get_isntance_generation(instance);` upon getting a MAP_FULL before releasing the lock for resize